### PR TITLE
Series.dropna scalable draft

### PR DESCRIPTION
**Pull request by 1e-to**
image

### CODE REVIEW

File: sdc/functions/numpy_like.py [hunk 1]
```diff
@@ -393,3 +395,103 @@ def nanprod_impl(a):
         return c
 
     return nanprod_impl
+
+
+def get_pool_size():
```
Comment (AlexanderKalistratov) [inline sdc/functions/numpy_like.py]: Let's move this file to something like `prange_utils.py`

File: sdc/functions/numpy_like.py [hunk 2]
```diff
@@ -393,3 +395,106 @@ def nanprod_impl(a):
...
+    def get_pool_size_impl():
+        return pool_size
+
+    return get_pool_size_impl
+
+def get_chunks(size, pool_size=0):
+    if pool_size == 0:
+        pool_size = get_pool_size()
+
+    chunk_size = size//pool_size + 1
```
Comment (AlexanderKalistratov) [inline sdc/functions/numpy_like.py]:
```suggestion
    chunk_size = (size - 1)//pool_size + 1
```

this is the correct formula

File: sdc/utilities/prange_utils.py [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+    return chunks
+
+
+@sdc_overload(get_chunks)
+def get_chunks_overload(size, pool_size=0):
+    def get_chunks_impl(size, pool_size=0):
+        if pool_size == 0:
+            pool_size = get_pool_size()
+
+        chunk_size = size//pool_size + 1
```
Comment (AlexanderKalistratov) [inline sdc/utilities/prange_utils.py]: `(size - 1)//pool_size + 1`

File: sdc/functions/numpy_like.py [hunk 4]
```diff
@@ -393,3 +395,51 @@ def nanprod_impl(a):
...
+                if not isnan(arr[j]):
+                    res += 1
+            length += res
+            arr_len[i] = res
+
+        result_data = numpy.empty(shape=length, dtype=dtype)
+        result_index = numpy.empty(shape=length, dtype=dtype_idx)
+        for i in prange(len(chunks)):
+            chunk = chunks[i]
+            if i == 0:
```
Comment (AlexanderKalistratov) [inline sdc/functions/numpy_like.py]:
```python
new_start = sum(arr_len[0:i])
new_stop = new_start + arr_len[i]
```

File: sdc/functions/numpy_like.py [hunk 5]
```diff
@@ -393,3 +395,51 @@ def nanprod_impl(a):
...
+                new_stop = arr_len[0]
+            else:
+                for x in range(i):
+                    new_start += arr_len[x]
+                new_stop = new_start + arr_len[i]
+
+            for j in range(chunk.start, chunk.stop):
+                if new_start < new_stop:
+                    if not isnan(arr[j]):
+                        result_data[new_start] = arr[j]
```
Comment (AlexanderKalistratov) [inline sdc/functions/numpy_like.py]: It is better to introduce new variable. Something like `current_pos`. Like this:
```python
current_pos = new_start
for j in range(chunk.start, chunk.stop):
    if current_pos < new_stop:
        if not isnan(arr[j]):
            result_data[current_pos] = arr[j]
            result_index[current_pos] = idx[j]
            current_pos += 1
```

It is confusing that you are always writing to `new_start`

File: sdc/functions/numpy_like.py [hunk 6]
```diff
@@ -393,3 +395,51 @@ def nanprod_impl(a):
...
+            if i == 0:
+                new_start = 0
+                new_stop = arr_len[0]
+            else:
+                for x in range(i):
+                    new_start += arr_len[x]
+                new_stop = new_start + arr_len[i]
+
+            for j in range(chunk.start, chunk.stop):
+                if new_start < new_stop:
```
Comment (AlexanderKalistratov) [inline sdc/functions/numpy_like.py]: Not sure if you actually need this condition

**Comment by AlexanderKalistratov** [comment]
conflicts

**Comment by AlexanderKalistratov** [comment]
Could you please remeasure performance?

**Comment by 1e-to** [comment]
> ase remeasure performance?

It's still like in a picture one:
image

**Pull request merged**